src/include/mpir_request.h

```diff
@@ -354,23 +364,23 @@ static inline MPIR_Request *MPIR_Request_create(MPIR_Request_kind_t kind, int po
 
 MPL_STATIC_INLINE_PREFIX MPIR_Request *MPIR_Request_create_complete(MPIR_Request_kind_t kind)
 {
-    MPIR_Request *req;
-
-#ifdef HAVE_DEBUGGER_SUPPORT
```

Comment:
We can't remove this debugger support.

Reply:
What does the debugger support need? A unique `MPIR_Request` pointer?

Reply:
Yes, one that includes the matching information (rank, tag, comm) for the posted send operation.

Reply:
Er, more than just send operations.

Reply:
Just worked on PR #4557 for `debuginfo` build. Seems we only support `sendq`, i.e. `SEND` and `PREQUEST_SEND` kinds, right? With builtin handle, I think we could add code in `dbginit.c` to special case the lightweight request. That way, we could isolate the debugger mess inside `dbginit.c`. Anyway, that will be a TODO in another time.

Reply:
Hmm, yes, that seems doable since the API already takes the matching information separately from the request ptr.